dvc/remote/ssh/connection.py

```diff
@@ -164,19 +164,24 @@ def download(self, src, dest, no_progress_bar=False, progress_title=None):
         os.rename(tmp_file, dest)
 
     def upload(self, src, dest, no_progress_bar=False, progress_title=None):
+        from uuid import uuid4
+
         self._sftp_connect()
 
         self.makedirs(posixpath.dirname(dest))
+        tmp_file = dest + "." + str(uuid4())
```

Comment:
Let's not duplicate code from RemoteBase.tmp_file(). Let's move it to `dvc/utils` and use it from there.